### PR TITLE
Fix `gr.Interface.from_pipeline()` to allow audio uploads and to display classification labels correctly

### PR DESCRIPTION
**Pull request by whitphx**
## Description

During #8052, I found something.

---
Reference: [first commit of `from_pipeline`](https://github.com/gradio-app/gradio/commit/c2f081e5bf7da7158ed92a2998df81fc61dbb4c3#diff-275f9a5e534203c28d9c5f44db604c6d0aa108005e22a118a8873a3c4af66d38)

### CODE REVIEW

File: gradio/pipelines_utils.py [hunk 1]
```diff
@@ -26,21 +26,14 @@ def is_transformers_pipeline_type(pipeline, class_name: str):
     # version of the transformers library that the user has installed.
     if is_transformers_pipeline_type(pipeline, "AudioClassificationPipeline"):
         return {
-            "inputs": components.Audio(
-                sources=["microphone"],
-                type="filepath",
-                label="Input",
-                render=False,
-            ),
+            "inputs": components.Audio(type="filepath", label="Input", render=False),
```
Comment (whitphx) [inline gradio/pipelines_utils.py:29]: Audio sources shouldn't be limited to "microphone"?

File: gradio/pipelines_utils.py [hunk 2]
```diff
@@ -26,21 +26,14 @@ def is_transformers_pipeline_type(pipeline, class_name: str):
...
             "outputs": components.Label(label="Class", render=False),
             "preprocess": lambda i: {"inputs": i},
-            "postprocess": lambda r: {i["label"].split(", ")[0]: i["score"] for i in r},
```
Comment (whitphx) [inline gradio/pipelines_utils.py:37]: This transforms a label name like "tiger, Panthera tigris" into just "tiger", which looks not good.

On `main`:
CleanShot 2024-04-19 at 16 27 16@2x

This change:
CleanShot 2024-04-19 at 16 27 50@2x

File: gradio/pipelines_utils.py [hunk 3]
```diff
@@ -66,7 +59,7 @@ def is_transformers_pipeline_type(pipeline, class_name: str):
             ),
             "outputs": components.Label(label="Classification", render=False),
             "preprocess": lambda i: {"images": i},
-            "postprocess": lambda r: {i["label"].split(", ")[0]: i["score"] for i in r},
```
Comment (whitphx) [inline gradio/pipelines_utils.py:69]: This transforms a label name like "tiger, Panthera tigris" into just "tiger", which looks not good.

On `main`:
CleanShot 2024-04-19 at 16 27 16@2x

This change:
CleanShot 2024-04-19 at 16 27 50@2x

Sample code:
```python
import gradio as gr
from transformers import pipeline


pipe = pipeline("image-classification", "google/vit-base-patch16-224")

demo = gr.Interface.from_pipeline(pipe)

demo.launch()
```

File: gradio/pipelines_utils.py (hunk 1 above)
Comment (abidlabs) [reply gradio/pipelines_utils.py:29]: Agreed

File: gradio/pipelines_utils.py (hunk 3 above)
Comment (abidlabs) [reply gradio/pipelines_utils.py:69]: Yeah in the general case better to leave the labels as is

**Review by abidlabs — APPROVED** [review]
lgtm @whitphx!

**Comment by abidlabs** [comment]
I would just suggest using a more precise name for this PR since that is what shows up in the changelog

**Comment by whitphx** [comment]
> that is what shows up in the changelog

Oh 👀 
Will do it.

**Pull request merged**